3d_armor/api.lua

```diff
@@ -933,6 +933,10 @@ armor.get_valid_player = function(self, player, msg)
 		minetest.log("warning", ("3d_armor%s: Player reference is nil"):format(msg))
 		return
 	end
+	if type(player) ~= "userdata" then
```

Comment:
You could use `not player:is_player()` in this place. Pipeworks and digtron implement this function, and you could add it by the excuse of "avoid calling this function on luaentities".

Reply:
No, that function returns true for pipeworks: https://github.com/mt-mods/pipeworks/blob/5e9bfa049b60805a73250dd495a4755f37247346/common.lua#L232

Interestingly digtron's fake player returns false, when it should be true, but really I'm not quite sure which is correct... I mean, they are players. 🤔

Reply:
I see. That's indeed a tricky situation. After all, checking against userdata seems to be the sanest solution even though it looks pretty hacky.